From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:13:48 +0530
Subject: [PATCH 013/109] Cygwin: define ___CTOR_LIST__ and
 ___DTOR_LIST__ for AArch64

(cherry picked from commit ef25935195339835d60948714cae16587a845770)
---
 winsup/cygwin/cygwin.sc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/cygwin.sc.in b/winsup/cygwin/cygwin.sc.in
index 4dc5daed8..63fa629de 100644
--- a/winsup/cygwin/cygwin.sc.in
+++ b/winsup/cygwin/cygwin.sc.in
@@ -21,7 +21,7 @@ SECTIONS
     *(SORT(.text$*))
     *(.glue_7t)
     *(.glue_7)
-#ifdef __x86_64__
+#if defined(__x86_64__) || defined(__aarch64__)
     . = ALIGN(8);
      ___CTOR_LIST__ = .; __CTOR_LIST__ = .;
 			LONG (-1); LONG (-1); *(SORT(.ctors.*)); *(.ctors); *(.ctor); LONG (0); LONG (0);
